src/Interpreters/Cache/FileCache.cpp

```diff
@@ -988,17 +988,18 @@ void FileCache::cleanup()
 
 void FileCache::cleanupThreadFunc()
 {
-#ifndef NDEBUG
-    assertCacheCorrectness();
-#endif
-
     try
     {
+#ifdef ABORT_ON_LOGICAL_ERROR
+        assertCacheCorrectness();
```

Comment:
Ok, we want to see clear exception message instead of `background task cannot throw`.